webdriver/tests/bidi/browsing_context/set_viewport/invalid.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+import webdriver.bidi.error as error
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+@pytest.mark.asyncio
+async def test_params_context_invalid_context(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.set_viewport(context=value, viewport={
+            "width": 100,
+            "height": 200,
+        })
+
+
+@pytest.mark.parametrize("viewport", [{"width": -1, "height": -1}, {"width": None, "height": 300}, {"width": 1200, "height": "String"}])
+@pytest.mark.asyncio
+async def test_params_context_invalid_viewport(bidi_session, new_tab, viewport):
```

Comment:
Mind differentiating between invalid type and invalid value? Also we have to test height and width in combination so that we know that also height alone set to `None` will fail. Other invalid tests of existing APIs should give a good sense how we did that.

The test name should then start with `test_params_viewport_invalid_` as well.

Reply:
Python's type checking didn't allow me to actually test things like missing height or missing width. Should I rewrite the set_viewport client implementation to now expect specific keys from the test?

Reply:
relaxed the client implementation and added more cases